zip-0213.rst

```diff
@@ -0,0 +1,179 @@
+transparent outputs. However, Zcash was launched with a consensus rule preventing coinbase
+transactions from containing shielded outputs, instead enforcing that coinbase funds could
+not be spent in transactions with transparent outputs. This was partly in order to reduce
+the complexity of the original Zcash modifications to the Bitcoin Core codebase, but also
+because at the time, shielded transactions required significant memory and CPU resources
+to create.
+
+The Sapling network upgrade [#zip-0205]_ deployed architectural changes and performance
+improvements that make shielding funds directly in the coinbase transaction feasible. In
+order to reduce the complexity of the Sapling network upgrade, the existing consensus
+rules preventing coinbase transactions from containing shielded outputs were extended to
+cover Sapling outputs. Therefore, it is now necessary to modify the consensus rules in
+order to enable miners to start using Sapling addresses.
+
+
+Specification
+=============
+
+Prior to activation of the TODO network upgrade, the existing consensus rule requiring
+coinbase transactions to have zero Sapling outputs is enforced.
+
+Once the TODO network upgrade activates:
+
+- The above existing consensus rule is no longer active, and coinbase transactions MAY
+  contain Sapling outputs.
+
+- The consensus rules applied to ``valueBalance``, ``vShieldedOutput``, and ``bindingSig``
+  in non-coinbase transactions MUST also be applied to coinbase transactions.
+
+- The existing consensus rule requiring transactions that spend coinbase outputs to have
+  an empty ``vout``, is amended to only apply to transactions that spend transparent
+  coinbase outputs.
+
+- The existing consensus rule requiring coinbase outputs to have 100 confirmations before
+  they may be spent (coinbase maturity), is amended to only apply to transparent coinbase
+  outputs.
+
+- All Sapling outputs in coinbase transactions MUST have valid note commitments when
+  recovered using a 32-byte array of zeroes as the outgoing viewing key.
+
+
+Interaction with funding streams
+--------------------------------
+
+If ZIP 207 [#zip-0207]_ is accepted and deployed prior to (or in the same network upgrade
+as) this ZIP, the following additional consensus rules are enforced once the TODO network
+upgrade activates:
+
+- Funding streams MAY target Sapling addresses in addition to transparent P2SH addresses.
+
+- The existing consensus rule from ZIP 207 [#zip-0207]_ requiring coinbase transactions
+  for block height ``height`` to pay active funding streams with a standard P2SH script,
+  is amended to only apply to active funding streams that target a transparent address at
+  ``height``.
+
+- In the coinbase transaction for block height ``height``, for every active funding stream
+  that targets a Sapling address at ``height``, the transaction MUST include at least one
+  Sapling output that pays exactly the funding stream's value to the target Sapling
+  address.
```

Comment:
Do you mean "exactly one Sapling output that pays exactly the funding stream's value to the target Sapling address." I think I understand what you meant - the transaction must have at least one output, but I think you might have literally said that it's okay to pay out the block reward multiple times in multiple outputs.

Reply:
What I'm saying is that it might be worth restating that the outputs of the coinbase transaction still have to balance.